### ISSUE TITLE
 exit status code '127' - No such file or directory

### ISSUE THREAD

**Issue by Mr-Anonymous** [barryvdh/laravel-snappy#177]
1. I have installed WKHTMLTOPDF already and have linked the binary to `/usr/bin/wkhtmltopdf` location

2. WKHTMLTOPDF Snappy was working fine until yesterday morning.

3. I am able to create pdf directly from using the WKHTMLTOPDF command.

4) The WKHTMLTOPDF binary exists in the `/usr/bin/` location and the correct path is already set in snappy config.

5) All the folders and files have the right permission assigned to the correct user who has the permission.

However, inspite of the above, I am receiving this error:

```
Caught exception: The exit status code '127' says something went wrong: stderr: "sh: /usr/bin/wkhtmltopdf: No such file or directory " stdout: "" command: /usr/bin/wkhtmltopdf --lowquality --dpi '150' --margin-bottom '20' --margin-left '10' --margin-right '10' --margin-top '10' --orientation 'Portrait' --page-size 'a4' --footer-html '/tmp/knp_snappy58b5e12b289ae8.79705067.html' --viewport-size '800x600' '/tmp/knp_snappy58b5e12b288d91.81452235.html' '/home/myapp/appfiles/laravel/downloads/my-filename-23-Feb-2017-1488314667.pdf'.
```

What is the error pointing too? The binary location `/usr/bin/wkhtmltopdf` is correct and the binary works using the command directly. I noticed the 2 files `/tmp/knp_snappy*` files and the `my-filename-23-Feb-2017-1488314667.pdf` file does not exists in those folders. Is that why its failing? Why is it throwing this error and where do I look at next to troubleshoot this?

I am totally stumped. Hopefully someone can guide me here.

Thanks,
Neel.

**Comment by smplyjr** [comment]
Hi @Mr-Anonymous, did you solved this? Can you share it? I'm facing this also now

**Comment by Mr-Anonymous** [comment]
Hi @smplyjr, For me it turned out to be due to php error in the blade file that creates the pdf. Checking the Failed Jobs in Database Table showed Undefined variable in PDF Blade file. I fixed this php error in my blade file. I also checked the Laravel Log and fixed all the errors I saw there to make sure there are no more errors in notifications or queue jobs, etc. In the end, I run all these commands to restart the queue, config and supervisor like this:
```
		php artisan queue:restart
		php artisan config:clear
		service supervisord restart
```
	
Doing the above fixed the issue and the queue started working. Hope this helps.
